StevesSixLeggedSprinters/BangazonAPI
# Employee Resource - Krissy

## Status
Choose:
**HOLD**

## Migrations
 NO
Migrations are going to have to be done locally.

## Description
This Pull Request Addresses Ticket #6 regarding the creation of an Employee Resource for developers creating the Bangazon Inc. Platform to reference. I have completed creating the Employee Model and The EmployeeController. The Controller will test the GET (List of current instances of Employees), POST (Adding a new Employee instance) and PUT (Ability to access and update the instance of Employee)to the bangazon.db. 

I also Changed the BangazonAPIContext file to include the route to the employee resource. Also with in this file I have added a ModelBuilder for Employee StartDate. This is currently set to insert the time stamp of the moment the record is created.  

This Resource has a dependency of the Department Resource creation prior to merging in the code. 


## Todos


## Deploy Notes
This is reliant on other dependencies, it is not yet ready for Deployment. It is only ready for team review. 

## Steps to Test or Reproduce

To Test Code, first pull down branch using the following Steps
```sh
git fetch --a
git branch -avv
git checkout <branch-name>
git pull origin <branch-name>
```
From there To test functionality. 

delete your Current bangazon.db 
AND 
Migrations folder.

go to the Model/Employee.cs and make sure that line 50 is commented out: 
```
public Department Department {get; set;}
``` 

```
dotnet ef migrations add Initial 
dotnet ef database update 
dotnet run
```
Make sure to have Postman installed: https://www.getpostman.com/
In Postman, 'Post' (add) a employees to the database to start testing.
```
{
	"EmpFirstName": "Krissy",
	"EmpLastName":"Caron",
	"EmpJobTitle" : "Software Developer",
}
```
The EmpStartDate and IsSupervisior fields will be auto generated on POST. 



## Impacted Areas in Application
This Resource has 1 dependency of the Deparment Resource and it will impact the 
- Employee Training Program Joint Resource also the Training Program Resource.
- Computers Resource

*
